lib/google_crawler_web/controllers/keyword_controller.ex

```diff
@@ -1,12 +1,16 @@
 defmodule GoogleCrawlerWeb.KeywordController do
   use GoogleCrawlerWeb, :controller
 
-  alias GoogleCrawler.{Repo, Search}
-  alias GoogleCrawler.Search.Keyword
+  alias GoogleCrawler.Repo
+  alias GoogleCrawler.Search
+  alias GoogleCrawler.Search.{Keyword, KeywordQueryParams}
 
-  def index(conn, _params) do
-    keywords = Search.list_keywords(conn.assigns[:current_user])
-    render(conn, "index.html", keywords: keywords)
+  def index(conn, params) do
+    %{query_params: query_params, changeset: changeset} =
+      KeywordQueryParams.new(params["filter"] || %{})
```

Comment:
I'm quite unsure about this 🤔 

Maybe you can take a look at the params validator on Toyota wallet? - https://github.com/nimblehq/gcp-api/blob/master/apps/gcpf_common/lib/gcpf_common/params_validator.ex. 

But you want to use both results - params and changeset here, right? 🤔  Then Toyota wallet's one might not fit here.

Reply:
@rosle Yes I need both the results here, one for easily rendering the form(changeset), as form_for helper supports only changeset, and another for filtering and passing to KeywordQuery.

Reply:
Aha, I see 👀  Actually for form_for, we can also use with atom like this `form_for :filter, ...` but we will need to populate the input value by ourself by, maybe `value: get_in(params, ["filter", "term"])`

Reply:
@rosle Yes I want to avoid that, that is the reason I went this way. :)